File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/freetrial/FreeTrialformuleCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.freetrial;

import java.util.List;
import ma.zs.alc.bean.core.freetrial.FreeTrialformule;
import ma.zs.alc.dao.criteria.core.freetrial.FreeTrialformuleCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface FreeTrialformuleCollaboratorService extends  IService<FreeTrialformule,FreeTrialformuleCriteria>  {

    List<FreeTrialformule> findByInscriptionId(Long id);
    int deleteByInscriptionId(Long id);
    long countByInscriptionId(Long id);




}

File: frontend-bqckup/src/app/shared/service/collaborator/freetrial/FreeTrialformuleCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {FreeTrialformuleDto} from '../../../model/freetrial/FreeTrialformule.model';
import {FreeTrialformuleCriteria} from '../../../criteria/freetrial/FreeTrialformuleCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class FreeTrialformuleCollaboratorService extends AbstractService<FreeTrialformuleDto, FreeTrialformuleCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/freeTrialformule/';
    }

    public constrcutDto(): FreeTrialformuleDto {
        return new FreeTrialformuleDto();
    }

    public constrcutCriteria(): FreeTrialformuleCriteria {
        return new FreeTrialformuleCriteria();
    }
}
